lib/train/extras/command_wrapper.rb

```diff
@@ -82,6 +82,8 @@ def verify
       when "sudo: sorry, you must have a tty to run sudo"
         ["Sudo requires a TTY. Please see the README on how to configure "\
           "sudo to allow for non-interactive usage.", :sudo_no_tty]
+      when /\[sudo\] password for root: Sorry, try again./
+        ["[sudo] password for root: Sorry, try again", :sudo_root_password_required]
```

Comment:
is there a need to distinguish between :sudo_root_password_required and :sudo_password_required and could this be made more generic by not matching the username explicitly in the prompt?

Reply:
Yes, both are raised in different scenarios for example:

1. if --sudo option is provided and it has configured in a way to required a `Sudo password` to perform any action then it raises eg.
`sudo_password_required`

```
Defaults requiretty 

OR

Defaults rootpw

```

2. if --sudo & --use-sudo-option both are provided then, so in this case, we also set the given user password in `opts[:sudo_password]` but if 
given password is not for root use then it raises 

:sudo_root_password_required


Matching the username here(root) is because it only raises in condition if we have set `Defaults rootpw` in the sudoers file

Reply:
Still feels like for train that this should be made generic and not match `root` specifically, and the label that comes back should not have "root" in it.  I see how `:sudo_password_required` is a very different error message, so it shouldn't reuse that label, but a better label should probably be used here.

Reply:
So, Probably, we could have a better name for `:sudo_password_required` if it matches `sudo: no tty present and no askpass program specified` but it too late to decide

Reply:
oh wait, we can match substring regex for `:bad_sudo_password`, as we attempt to login using the current user password

Reply:
```
case rawerr
when \Sorry, try again\
["Wrong sudo password.", :bad_sudo_password]
```

Is covering our purpose?

Reply:
ah there we go, this makes much more sense now.